server/src/main/java/io/druid/server/listener/resource/AbstractListenerHandler.java

```diff
@@ -0,0 +1,221 @@
+    } else {
+      return Response.status(Response.Status.ACCEPTED).entity(returnObj).build();
+    }
+  }
+
+  @Override
+  @NotNull
+  public final Response handleGET(@NotNull String id)
+  {
+    try {
+      final Object returnObj = get(id);
+      if (returnObj == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      } else {
+        return Response.ok(returnObj).build();
+      }
+    }
+    catch (Exception e) {
+      LOG.error(e, "Error handling get request for [%s]", id);
+      return Response.serverError().entity(ServletResourceUtils.sanitizeException(e)).build();
+    }
+  }
+
+  @Override
+  @NotNull
+  public final Response handleGETAll()
+  {
+    final Collection<ObjType> all;
+    try {
+      all = getAll();
+      if (all == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      } else {
+        return Response.ok(all).build();
+      }
+    }
+    catch (Exception e) {
+      LOG.error(e, "Error getting all");
+      return Response.serverError().entity(ServletResourceUtils.sanitizeException(e)).build();
+    }
+  }
+
+  @Override
+  @NotNull
+  public final Response handleDELETE(@NotNull String id)
+  {
+    try {
+      final Object returnObj = delete(id);
+      if (returnObj == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      } else {
+        return Response.status(Response.Status.ACCEPTED).entity(returnObj).build();
+      }
+    }
+    catch (Exception e) {
+      LOG.error(e, "Error in processing delete request for [%s]", id);
+      return Response.serverError().entity(ServletResourceUtils.sanitizeException(e)).build();
+    }
+  }
+
+  @Override
+  public final void use_AbstractListenerHandler_instead()
+  {
+    // NOOP
+  }
+
+  /**
+   * Delete the object for a particular id
+   *
+   * @param id A string id of the object to be deleted. This id is never null or empty.
+   *
+   * @return The object to be returned in the entity. A NULL return will cause a 404 response. A non-null return will cause a 202 response. An Exception thrown will cause a 500 response.
+   */
+  protected abstract
+  @Nullable
+  Object delete(@NotNull String id);
+
+  /**
+   * Get the object for a particular id
+   *
+   * @param id A string id of the object desired. This id is never null or empty.
+   *
+   * @return The object to be returned in the entity. A NULL return will cause a 404 response. A non-null return will cause a 200 response. An Exception thrown will cause a 500 response.
+   */
+  protected abstract
+  @Nullable
+  Object get(@NotNull String id);
+
+  protected abstract
+  @Nullable
+  Collection<ObjType> getAll();
+
+  /**
+   * Process a POST request of the input items
+   *
+   * @param inputObject A list of the objects which were POSTed
+   *
+   * @return An object to be returned in the entity of the response.
+   *
+   * @throws Exception
+   */
+  public abstract
+  @Nullable
```

Comment:
I put it in here to be explicit about which methods can return null and which ones cannot